rsvim_core/src/state/fsm/normal_tests.rs

```diff
@@ -8091,3 +8091,230 @@ mod tests_goto_command_line_ex_mode {
+
+  use crossterm::event::{Event, KeyCode, KeyEvent, KeyEventKind, KeyModifiers};
+  use std::collections::BTreeMap;
+  use tokio::sync::mpsc::{Receiver, Sender, channel};
+  use tracing::info;
+
+  #[test]
+  fn nowrap1_go_to_normal_insert() {
+    test_log_init();
+
+    let terminal_size = U16Size::new(30, 3);
+    let (tree, state, bufs, _buf, contents) = make_tree(
+      terminal_size,
+      WindowLocalOptionsBuilder::default()
+        .wrap(false)
+        .build()
+        .unwrap(),
+      vec!["Should go to insert mode\n"],
+    );
+
+    let key_event = KeyEvent::new_with_kind(
+      KeyCode::Char('a'),
+      KeyModifiers::empty(),
+      KeyEventKind::Press,
+    );
+
+    let prev_cursor_viewport = get_cursor_viewport(tree.clone());
+    assert_eq!(prev_cursor_viewport.line_idx(), 0);
+    assert_eq!(prev_cursor_viewport.char_idx(), 0);
+
+    let data_access = StatefulDataAccess::new(state, tree, bufs, contents, Event::Key(key_event));
+    let stateful = NormalStateful::default();
+    stateful.cursor_move(&data_access, Operation::CursorMoveRightBy(2));
+    let insert_result =
+      stateful.goto_insert_mode(&data_access, crate::state::ops::InsertMotion::Normal);
+    assert_eq!(
+      insert_result,
+      StatefulValue::InsertMode(InsertStateful::default())
+    );
+
+    let tree = data_access.tree.clone();
+    let actual_cursor = get_cursor_viewport(tree.clone());
+    assert_eq!(actual_cursor.line_idx(), 0);
+    assert_eq!(actual_cursor.char_idx(), 2);
+    assert_eq!(actual_cursor.row_idx(), 0);
+    assert_eq!(actual_cursor.column_idx(), 2);
+
+    let expect_canvas = vec![
+      "Should go to insert mode      ",
+      "                              ",
+      "                              ",
+    ];
+    let actual_canvas = make_canvas(tree.clone(), terminal_size);
+    let actual_canvas = lock!(actual_canvas);
+    assert_canvas(&actual_canvas, &expect_canvas);
+  }
+
+  #[test]
+  fn nowrap2_go_to_append_insert() {
+    test_log_init();
+
+    let terminal_size = U16Size::new(30, 3);
+    let (tree, state, bufs, _buf, contents) = make_tree(
+      terminal_size,
+      WindowLocalOptionsBuilder::default()
+        .wrap(false)
+        .build()
+        .unwrap(),
+      vec!["Should go to insert mode\n"],
+    );
+
+    let key_event = KeyEvent::new_with_kind(
+      KeyCode::Char('a'),
+      KeyModifiers::empty(),
+      KeyEventKind::Press,
+    );
+
+    let prev_cursor_viewport = get_cursor_viewport(tree.clone());
+    assert_eq!(prev_cursor_viewport.line_idx(), 0);
+    assert_eq!(prev_cursor_viewport.char_idx(), 0);
+
+    let data_access = StatefulDataAccess::new(state, tree, bufs, contents, Event::Key(key_event));
+    let stateful = NormalStateful::default();
+    stateful.cursor_move(&data_access, Operation::CursorMoveRightBy(2));
+    let insert_result =
+      stateful.goto_insert_mode(&data_access, crate::state::ops::InsertMotion::Append);
```

Comment:
It would be better, if this test do below steps:
1. It first call "goto_insert_mode" with "Append"
2. It inserts some text payload.
3. Verify after inserted, the text payload is expected.

But no worry, I can merge the code and update these test cases.